### DISCUSSION TITLE
get secrets from docker swarm

### DISCUSSION THREAD

**Discussion by smeana** [quarkusio/quarkus#23810] [Q&A]
Hello 

Migrating from Sping where we read secrets like `spring.datasource.user={{ secret "db_user" }}`

Using Quarkus, what is the best way to pass secrets from docker swarm into the application.properties?

Regards

**Comment by edufolly** [comment]
Hi @smeana

I faced the same question and found [SmallRye FileSystem Config Source](https://smallrye.io/docs/smallrye-config/main/config-sources/config-sources.html#filesystem-config-source).

I created a PoC repo to check how it works. I hope that help you. https://github.com/edufolly/work-with-secrets/tree/prod

Best.
